components/datadog/operator/operator.go

```diff
@@ -0,0 +1,46 @@
+package operator
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/datadog/operatorparams"
+)
+
+// OperatorOutput is used to import the Operator component
+type Output struct {
+	components.JSONImporter
+}
+
+// Operator represents an Operator installation
+type Operator struct {
+	pulumi.ResourceState
+	components.Component
+}
+
+func (h *Operator) Export(ctx *pulumi.Context, out *Output) error {
+	return components.Export(ctx, h, out)
+}
+
+func NewOperator(e config.Env, resourceName string, kubeProvider *kubernetes.Provider, options ...operatorparams.Option) (*Operator, error) {
+	return components.NewComponent(e, resourceName, func(_ *Operator) error {
+		params, err := operatorparams.NewParams(e, options...)
+		if err != nil {
+			return err
+		}
+
+		_, err = NewHelmInstallation(e, HelmInstallationArgs{
+			KubeProvider:          kubeProvider,
+			Namespace:             params.Namespace,
+			ValuesYAML:            params.HelmValues,
+			OperatorFullImagePath: params.OperatorFullImagePath,
+		}, params.PulumiResourceOptions...)
```

Comment:
Can you append `pulumi.Parent(comp)` to the PulumiResourceOptions list and pass it to `NewHelmInstallation`. 
To tell Pulumi that the Helm component you create in `NewHelmInstallation` is a child of the `Operator` component. Otherwise we will end up with weird Pulumi dependencies.
`comp` is `comp *Operator` that you currently replaced with `_ *Operator`